cmd/boot-script-service/default_api.go

```diff
@@ -994,42 +995,65 @@ func DumpstateGet(w http.ResponseWriter, r *http.Request) {
 	var results State
 	state := getState()
 	results.Components = state.Components
-	for _, image := range GetKernelInfo() {
-		var bp bssTypes.BootParams
-		bp.Params = image.Params
-		bp.Kernel = image.Path
-		results.Params = append(results.Params, bp)
-	}
-	for _, image := range GetInitrdInfo() {
-		var bp bssTypes.BootParams
-		bp.Params = image.Params
-		bp.Initrd = image.Path
-		results.Params = append(results.Params, bp)
-	}
+	var err error
+	if useSQL {
+		results.Params, err = bssdb.GetBootParamsAll()
+		if err != nil {
+			log.Printf("DumpStateGet(): GetBootParamsAll(): Could not get boot parameters from SQL DB: %v", err)
+			err = fmt.Errorf("Error retrieving boot parameters from database")
+		}
+	} else {
+		for _, image := range GetKernelInfo() {
+			var bp bssTypes.BootParams
+			bp.Params = image.Params
+			bp.Kernel = image.Path
+			results.Params = append(results.Params, bp)
+		}
+		for _, image := range GetInitrdInfo() {
+			var bp bssTypes.BootParams
+			bp.Params = image.Params
+			bp.Initrd = image.Path
+			results.Params = append(results.Params, bp)
+		}
 
-	kvl, err := getTags()
-	var names []string
-	if err == nil {
-		for _, x := range kvl {
-			name := extractParamName(x)
-			names = append(names, name)
-			var bds BootDataStore
-			if e := json.Unmarshal([]byte(x.Value), &bds); e == nil {
-				bd := bdConvert(bds)
-				var bp bssTypes.BootParams
-				bp.Hosts = append(bp.Hosts, name)
-				bp.Params = bd.Params
-				bp.Kernel = bd.Kernel.Path
-				bp.Initrd = bd.Initrd.Path
-				results.Params = append(results.Params, bp)
+		var (
+			kvl   []hmetcd.Kvi_KV
+			names []string
+		)
+		kvl, err = getTags()
+		if err == nil {
+			for _, x := range kvl {
+				name := extractParamName(x)
+				names = append(names, name)
+				var bds BootDataStore
+				if e := json.Unmarshal([]byte(x.Value), &bds); e == nil {
+					bd := bdConvert(bds)
+					var bp bssTypes.BootParams
+					bp.Hosts = append(bp.Hosts, name)
+					bp.Params = bd.Params
+					bp.Kernel = bd.Kernel.Path
+					bp.Initrd = bd.Initrd.Path
+					results.Params = append(results.Params, bp)
+				} else {
+					debugf("WARNING: Unmarshalling boot data store for name %q and tag %v failed (not including in results): %v", name, x, err)
+				}
 			}
+			debugf("Retrieved names: %v", names)
+			debugf("Retrieved params: %v", results.Params)
+		} else {
+			log.Printf("DumpStateGet(): getTags(): %v", err)
+			err = fmt.Errorf("Error retrieving names from key-value store")
 		}
 	}
-	debugf("Retreived names: %v", names)
-	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
-	w.WriteHeader(http.StatusOK)
-	err = json.NewEncoder(w).Encode(results)
 	if err != nil {
```

Comment:
Where is the function producing this error now? It looks like the old code was moved into the `else` branch below.

Reply:
> the old code was moved into the else branch below.

Correct. The `else` block performs the actions for EtcD whereas the `if` block performs the actions for Postgres. `err` is declared beforehand and is set in each of these blocks (see reply to quote below), depending on which backend is being used.

> Where is the function producing this error now?

See [here](https://github.com/OpenCHAMI/bss/pull/42/files/bb7c3b31f7e0b705edf3c2841f967700dd0d1b4f#diff-b58ae3c469d8de26ccac04652baa29944e23c39b84ee0538e7f706c6d007c81dR1003) and [here](https://github.com/OpenCHAMI/bss/pull/42/files/bb7c3b31f7e0b705edf3c2841f967700dd0d1b4f#diff-b58ae3c469d8de26ccac04652baa29944e23c39b84ee0538e7f706c6d007c81dR1045). The `err` variable is declared beforehand [here](https://github.com/OpenCHAMI/bss/pull/42/files/bb7c3b31f7e0b705edf3c2841f967700dd0d1b4f#diff-b58ae3c469d8de26ccac04652baa29944e23c39b84ee0538e7f706c6d007c81dR998) so that it persists to be checked at this block.

Reply:
Ah, it make sense now. I don't think I've seen errors handle like that thus far..

Reply:
If there is a more idomatic way of doing it, I'm open to suggestions. I tend to be one for not redeclaring `err` every time it's used. 😉 

However, it's worth pointing out that there are a couple of spots where the value of `err` is printed in the logs, then immediately after it is set to a different error that is sent in the JSON payload. See [here](https://github.com/OpenCHAMI/bss/pull/42/files#diff-b58ae3c469d8de26ccac04652baa29944e23c39b84ee0538e7f706c6d007c81dR1044-R1045) and [here](https://github.com/OpenCHAMI/bss/pull/42/files#diff-b58ae3c469d8de26ccac04652baa29944e23c39b84ee0538e7f706c6d007c81dR1002-R1003). I don't foresee this being problematic, do you?

Reply:
It shouldn't be problematic, but I think that's the first time I've seen handle like that. Then again, I was never really a fan of Go's error handling in general and wished there was a more intuitive convention for it considering how much it's emphasized.

Reply:
Yeah I can see that. I think that this should work for now unless any subsequent issues come up. What do you think?

Reply:
Yep, I think we're good here.